<a href="https://colab.research.google.com/github/sindhu692/sindhu.py/blob/main/Power_usage_classifier.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [7]:
import numpy
import numpy as np
l=[]
for i in range(7):
  x=float(input(f"Enter your 7 days power Day {i+1}: "))
  l.append(x)
arr=np.array(l)
print(arr)
def classify_power(usage):
  low_days=0
  moderate_days=0
  high_days=0
  for i in range(len(usage)):
    if usage[i]<2.5:
      low_days+=1
    elif usage[i]>=2.5 and usage[i]<=5:
      moderate_days+=1
    else:
      high_days+=1
  print("Summary:")
  print(f"Low: {low_days} day(s)")
  print(f"Moderate: {moderate_days} day(s)")
  print(f"High: {high_days} day(s)")
classify_power(arr)


Enter your 7 days power Day 1: 2.3
Enter your 7 days power Day 2: 2.5
Enter your 7 days power Day 3: 2.8
Enter your 7 days power Day 4: 3.5
Enter your 7 days power Day 5: 6.7
Enter your 7 days power Day 6: 6.6
Enter your 7 days power Day 7: 8.9
[2.3 2.5 2.8 3.5 6.7 6.6 8.9]
Summary:
Low: 1 day(s)
Moderate: 3 day(s)
High: 3 day(s)


Project: Power Usage Classifier
🧾 Goal:
Given a NumPy array of daily power usage values (in kWh), we want to classify each day's usage into:

    🔵 Low → Usage < 2.5

    🟡 Moderate → Usage between 2.5 and 5 (inclusive of 2.5, but not 5)

    🔴 High → Usage ≥ 5

✅ Step-by-Step Logic Explanation

🔹 Step 1:

    Import NumPy & Create Input Array
    
    You’ll use a NumPy array to store power usage for 7 days (1 week).

    Example: [2.1, 3.5, 4.2, 6.0, 1.8, 5.6, 3.3]

    Each index = one day (Day 1 to Day 7)

🔹 Step 2:
    
    Create a Function

    Define a function like classify_power(usage)

It should:

    Loop through each value in the usage array

Based on value:

    Label it as Low, Moderate, or High

    Print the label with corresponding day

🔹 Step 3:

    Inside the Loop → Use Conditions
    
    For each day's value:

    If value < 2.5 → mark as “Low”

    Else if value between 2.5 and 5 → “Moderate”

    Else → “High”

👉 These conditions can be done using:

    A simple if-elif-else structure

    Or a cleaner way using NumPy's where (optional at intermediate level)

🔹 Step 4:
    
    Count the Categories
   
    While you’re classifying each day, also maintain counters for:

    low_days, moderate_days, high_days

    Increment them as per classification.

🔹 Step 5:
    
    3Print Final Summary

    After the loop:

Print:
Summary:

    Low: X days
    Moderate: Y days
    High: Z days
This gives a weekly overview of power usage.